kolibri/core/assets/src/views/HorizontalNavBarWithOverflowMenu.vue

```diff
@@ -0,0 +1,151 @@
+        :title="link.title"
+        :link="link.link"
+      >
+        <KIcon
+          :icon="link.icon"
+          :color="link.color"
+        />
+      </NavbarLink>
+    </div>
+    <span v-if="overflowMenuLinks && overflowMenuLinks.length > 0">
+      <KIconButton
+        class="menu-icon"
+        :tooltip="coreString('moreOptions')"
+        tooltipPosition="top"
+        :ariaLabel="coreString('moreOptions')"
+        icon="optionsHorizontal"
+        appearance="flat-button"
+        :color="color"
+        :primary="false"
+      >
+        <template #menu>
+          <KDropdownMenu
+            :primary="false"
+            :disabled="false"
+            :hasIcons="true"
+            :options="overflowMenuLinks"
+            @select="handleSelect"
+          />
+        </template>
+      </KIconButton>
+    </span>
+  </Navbar>
+
+</template>
+
+
+<script>
+
+  import Navbar from 'kolibri.coreVue.components.Navbar';
+  import NavbarLink from 'kolibri.coreVue.components.NavbarLink';
+  import commonCoreStrings from 'kolibri.coreVue.mixins.commonCoreStrings';
+  import responsiveWindowMixin from 'kolibri.coreVue.mixins.responsiveWindowMixin';
+  import debounce from 'lodash/debounce';
+
+  export default {
+    name: 'HorizontalNavBarWithOverflowMenu',
+    components: {
+      Navbar,
+      NavbarLink,
+    },
+    mixins: [commonCoreStrings, responsiveWindowMixin],
+    props: {
+      /**
+       * An array of options objects, with one object per dropdown item
+       */
+      navigationLinks: {
+        type: Array,
+        default: () => [],
+        required: true,
+      },
+    },
+    data() {
+      return {
+        numberOfNavigationTabsToDisplay: 0,
+        overflowMenuLinks: [],
+      };
+    },
+    computed: {
+      color() {
+        return this.$themeTokens.textInverted;
+      },
+    },
+    mounted() {
+      this.updateNavigationTabDisplay();
+      window.addEventListener('resize', this.updateNavigationTabDisplay);
+      // document.addEventListener('resize', console.log('hello world'));
+    },
+    beforeDestroy() {
+      document.removeEventListener('resize', this.debouncedUpdateNavigation);
+    },
+    methods: {
+      debouncedUpdateNavigation() {
+        return debounce(this.updateNavigationTabDisplay, 1000);
+      },
+      handleSelect(option) {
+        this.$router.push(this.$router.getRoute(option.value.name));
+      },
+      generateOverflowMenu() {
+        const limitedList = this.navigationLinks.slice(
+          this.numberOfNavigationTabsToDisplay,
+          this.navigationLinks.length
+        );
+        let options = [];
+        limitedList.forEach(o => options.push({ label: o.title, value: o.link, icon: o.icon }));
+        this.overflowMenuLinks = options;
+      },
+      updateNavigationTabDisplay() {
+        // turn HTML collection into an array
+        const navItems = [].slice.call(document.getElementsByClassName('list-item-navigation'));
```

Comment:
Accessing `NavbarLinks` via classes in this way creates a hidden dependency that makes things quite brittle, e.g. if someone changes the class name in `NavbarLink`, this whole logic would silently break. I think that people usually don't get an idea to search whether a class from a component is used in a parent of its parent (`HorizontalNavBarWithOverflowMenu` -> `Navbar` -> `NavbarLink`). Perhaps `ref`s could be utilized? I'm not sure if that's the best idea, or if it'd be better to consider a different architecture, we can think about it more together if it'd be helpful. Any other ideas?

Reply:
`refs` is a good idea -- I will try implementing this way, and if I run into a problem today, I'll comment more here or in slack. And thank you for the suggestion and feedback, this is a good point. Even when I was working on it, it seemed a bit... hacky, but I didn't revisit a better approach after "getting it to work". But I should 😄

Reply:
Thank you, it's looking good to me, and I have no further ideas